### DISCUSSION TITLE
How can I draw an svg file to a rectangle in TImage32?

### DISCUSSION THREAD

**Discussion by tomwiel** [AngusJohnson/Image32#50] [General]
Graphics APIs typically use a definable world transform (affine matrix) and then draw objects.
The caller provides a matrix (optionally), and later draws a picture object (svg) to a destination rectangle (typically has svg aspect ratio). 

I've found SvgReader.DrawImage(), but this cannot work for this purpose, but img size should remain unchanged, SvgReader.RootElement (picture object) needs to know the world transform, which shall transform the destination rectangle (and svg).

Edited: It could be possible with a modified version
TSvgReader.DrawImage2( img: TImage32; const parentTransform: TMatrixD; const dstRect: TRectD); 
This one preserves the incoming img size, and di.matrix is created by more parameters (parentTransform, viewbox-to-dstRect)

**Comment by AngusJohnson** [comment]
1. Draw the SVG image to a background TImage32 object. (If the TImage32 object is empty (height & width = 0) then the SVG will be drawn using the SVG's default size, otherwise by presetting the object's height and width the SVG will scale proportionally into that size.)
2. Copy or blend the desired region/rectangle of that object (ie of the rasterized SVG image) into the specified rectangle in your main/foreground TImage32 object.
